src/commands/testing/test.ts

```diff
@@ -258,15 +180,89 @@ export async function runTest(
         authConfig,
         ['log']
       )
-      .then(handleRes)
-      .catch(async (err) => {
-        if (err.error?.message === 'Error: invalid Json string') {
-          await handleRes({})
-          return
+      .then(async (res: any) => {
+        let lineBreak = true
+
+        const test_results = res.result?.test_results
+
+        if (!test_results) {
+          displayError(
+            {},
+            `Job did not return a response, to debug click ${testUrl}`
+          )
+
+          printCodeExample()
+
+          lineBreak = false
+
+          const existingTestTarget = result.sasjs_test_meta.find(
+            (testResult: TestDescription) =>
+              testResult.test_target === testTarget
+          )
+
+          if (existingTestTarget) {
+            existingTestTarget.results.push({
+              test_loc: test,
+              sasjs_test_id: testId,
+              result: TestResultStatus.notProvided,
+              test_url: testUrl
+            })
+          } else {
+            result.sasjs_test_meta.push({
+              test_target: testTarget,
+              results: [
+                {
+                  test_loc: test,
+                  sasjs_test_id: testId,
+                  result: TestResultStatus.notProvided,
+                  test_url: testUrl
+                }
+              ]
+            })
+          }
+        }
+
+        if (res.log) await saveLog(outDirectory!, test, res.log, lineBreak)
+
+        if (test_results) {
+          const existingTestTarget = result.sasjs_test_meta.find(
+            (testResult: TestDescription) =>
+              testResult.test_target === testTarget
+          )
+
+          if (existingTestTarget) {
+            existingTestTarget.results.push({
+              test_loc: test,
+              sasjs_test_id: testId,
+              result: test_results,
+              test_url: testUrl
+            })
+          } else {
+            result.sasjs_test_meta.push({
+              test_target: testTarget,
+              results: [
+                {
+                  test_loc: test,
+                  sasjs_test_id: testId,
+                  result: test_results,
+                  test_url: testUrl
+                }
+              ]
+            })
+          }
+        } else if (target.serverType !== ServerType.Sasjs) {
+          displayError(
+            {},
+            `'test_results' not found in server response, to debug click ${testUrl}`
+          )
+
+          printCodeExample()
         }
+      })
+      .catch(async (err) => {
         printTestUrl()
```

Comment:
This major change is nothing, just converted named callback function `handleRes` to unnamed for `.then`.

Reply:
Can you provide a line of explanation about why you changed `handleRes` callback function to unnamed in then block?

Reply:
It was redundant, to have named callback function for then.

It was added by me few days ago( in order to have similar behaviour for SASJS) , in order to handle a case, but we configured adapter to return correct response.